geopandas/vectorized.pyx

```diff
@@ -0,0 +1,1029 @@
+    cdef bytearray out = bytearray()
+
+    handle = get_geos_context_handle()
+
+    writer = GEOSWKBWriter_create_r(handle)
+
+    for idx in xrange(n):
+        geos_geom = geoms[idx]
+        geom = <GEOSGeometry *> geos_geom
+        if geom:
+            c_string = GEOSWKBWriter_write_r(handle, writer, geom, &size)
+            py_string = c_string[:size]
+            out.extend(py_string)
+            free(c_string)
+            sizes[idx] = size
+        else:
+            sizes[idx] = 0
+
+    GEOSWKBWriter_destroy_r(handle, writer)
+    return out, sizes
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cpdef deserialize(const unsigned char* data, np.ndarray[np.uintp_t, ndim=1, cast=True] sizes):
+    cdef Py_ssize_t idx
+    cdef GEOSContextHandle_t handle
+    cdef GEOSGeometry *geom
+    cdef uintptr_t geos_geom
+    cdef unsigned int n = sizes.size
+    cdef size_t size
+    cdef GEOSWKBReader *reader
+
+    cdef np.ndarray[np.uintp_t, ndim=1] out = np.empty(n, dtype=np.uintp)
+
+    handle = get_geos_context_handle()
+
+    reader = GEOSWKBReader_create_r(handle)
+
+    with nogil:
+        for idx in xrange(n):
+            size = sizes[idx]
+            if size:
+                geom = GEOSWKBReader_read_r(handle, reader, data, size)
+                out[idx] = <np.uintp_t> geom
+            else:
+                out[idx] = 0
+            data += size
+
+    GEOSWKBReader_destroy_r(handle, reader)
+    return out
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cdef vec_free(np.ndarray[np.uintp_t, ndim=1, cast=True] geoms):
+    cdef Py_ssize_t idx
+    cdef GEOSContextHandle_t handle
+    cdef GEOSGeometry *geom
+    cdef uintptr_t geos_geom
+    cdef unsigned int n = geoms.size
+
+    handle = get_geos_context_handle()
+
+    with nogil:
+        for idx in xrange(n):
+            geos_geom = geoms[idx]
+            geom = <GEOSGeometry *> geos_geom
+            if geom:
+                GEOSGeom_destroy_r(handle, geom)
+
+
+class VectorizedGeometry(object):
+    dtype = np.dtype('O')
+
+    def __init__(self, data, parent=False):
+        self.data = data
+        self.parent = parent
+
+    def __getitem__(self, idx):
+        if isinstance(idx, numbers.Integral):
+            return get_element(self.data, idx)
+        elif isinstance(idx, (collections.Iterable, slice)):
+            return VectorizedGeometry(self.data[idx], parent=self)
+        else:
+            raise TypeError("Index type not supported", idx)
+
+    def take(self, idx):
+        result = self[idx]
+        result.data[idx == -1] = 0
+        return result
+
+    def __len__(self):
+        return len(self.data)
+
+    @property
+    def size(self):
+        return len(self.data)
+
+    def __del__(self):
+        if self.parent is False:
+            vec_free(self.data)
+
+    def copy(self):
+        # TODO dummy implementation just to have someting
+        return from_shapely(to_shapely(self.data))
```

Comment:
Yeah, I added this one some time ago because I just needed a `copy` method on this class to have the GeometryBlock working, but this needs a better implementation (I suppose using the GEOSGeom_clone_r)